File: 1023/1023.ts
function camelMatch(queries: string[], pattern: string): boolean[] {
    const ans: boolean[] = [];
    for (const query of queries) {
        ans.push(isMatch(query, pattern));
    }
    return ans;
}

function isMatch(query: string, pattern: string): boolean {
    let j = 0;
    for (const c of query) {
        if (j < pattern.length && c === pattern[j]) {
            ++j;
        } else if (c.toUpperCase() === c) {
            return false;
        }
    }
    return j === pattern.length;
}

File: Solutions/1023/1023.go
func camelMatch(queries []string, pattern string) []bool {
	var ans []bool
	for _, query := range queries {
		ans = append(ans, isMatch(query, pattern))
	}
	return ans
}

func isMatch(query string, pattern string) bool {
	j := 0
	for _, c := range query {
		if j < len(pattern) && byte(c) == pattern[j] {
			j++
		} else if unicode.IsUpper(c) {
			return false
		}
	}
	return j == len(pattern)
}